cmd/registry/conformance/conformance-task.go

```diff
@@ -162,14 +202,17 @@ func (task *ComputeConformanceTask) computeConformanceReport(
 
 				// Check if the guideline report for the guideline which contains this rule
 				// has already been initialized. If it hasn't then create one.
-				guidelineReport, ok := guidelineReportsMap[guideline.GetId()]
+				reportIndex, ok := guidelineReportsMap[guideline.GetId()]
```

Comment:
`guidelineReportsMap` now stores only the index of the `guidelineReport` in the `guidelineGroup` and not the whole object itself.

Example:
```
conformanceReport{
    ...
    GuidelineReportGroups: []*rpc.GuidelineReportGroup{
         &rpc.GuidelineReportGroup{...Status:UNSPECIFIED...},
         &rpc.GuidelineReportGroup{...Status:PROPOSED...},
         &rpc.GuidelineReportGroup{
             ...
             Status: "ACTIVE",
             GuidelineReports: []*rpc.GuidelineReport{
                 &rpc.GuidelineReport{GuidelineId: "test-guideline-id-1", ...}},
                 &rpc.GuidelineReport{GuidelineId: "test-guideline-id-2", ...}},
             ...
        }
    }
}
```

The `guidelineReportsMap` will store the index and not the whole `guidelineReport` object. The map will look like this for the above example:
```
"test-guideline-id-1": 0
"test-guideline-id-2": 1
```
We need to know the position of that particular `guidelineReport` in the corresponding status group. Then append the new `RuleReports` under that particular guideline.